Src/BootCamp.Chapter/Checks.cs

```diff
@@ -9,31 +7,192 @@ public static class Checks
     {
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (Testers.IsThisStringArrayValid(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+            string poorestName = "";
+            decimal amount = 0;
+            int personCount = 1;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var a = new Account(peopleAndBalances[i]);
+                if (i == 0)
+                {
+                    poorestName = a.GetName();
+                    amount = a.CurrentBalance();
+                }
+                else if (amount == a.CurrentBalance())
+                {
+                    if (personCount == 1)
+                    {
+                        poorestName += " and " + a.GetName();
+                        personCount++;
+                    }
+                    else
+                    {
+                        poorestName = poorestName.Replace(" and ", ", ");
+                        poorestName += " and " + a.GetName();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5123803)